lifecycle-operator/controllers/lifecycle/schedulinggates/controller_test.go

```diff
@@ -335,3 +335,29 @@ func TestSchedulingGatesReconciler_Reconcile(t *testing.T) {
 		})
 	}
 }
+
+func TestHasKeptnSchedulingGate(t *testing.T) {
+	t.Run("PodWithSchedulingGate", func(t *testing.T) {
+		pod := &v1.Pod{
+			Spec: v1.PodSpec{
+				SchedulingGates: []v1.PodSchedulingGate{
+					{
+						Name: apicommon.KeptnGate,
+					},
+				},
+			},
+		}
+
+		hasGate := hasKeptnSchedulingGate(pod)
+
+		require.True(t, hasGate)
+	})
+
+	t.Run("PodWithoutSchedulingGate", func(t *testing.T) {
+		podNoGate := &v1.Pod{}
+
+		hasGate := hasKeptnSchedulingGate(podNoGate)
+
+		require.False(t, hasGate)
+	})
```

Comment:
let's rewrite this as a table test and cover more scenarios, for example we have more gates present in the pod than only the keptn one, we have other gates present, but not the keptn one etc

Reply:
I have made the suggested change @odubajDT